src/community/wps-download/src/main/java/org/geoserver/wps/gs/download/RasterDownload.java

```diff
@@ -510,14 +506,34 @@ private GridCoverage2D readAndReproject(
                     LOGGER.log(Level.FINE, "Reprojecting the coverage");
                 }
                 disposableSources.add(gridCoverage);
-                gridCoverage =
+                GridCoverage2D testCoverage =
                         (GridCoverage2D)
                                 Operations.DEFAULT.resample(
                                         gridCoverage,
                                         targetCRS,
                                         null,
                                         interpolation,
                                         backgroundValues);
+                GridGeometryProvider gridGeometryProvider =
```

Comment:
I think that this path should be executed only if resolutionDifferenceTolerance is not zero since it's an optional parameter/for optional behavior. Does it make sense?